ocs_ci/deployment/helpers/external_cluster_helpers.py

```diff
@@ -187,7 +198,50 @@ def get_rgw_endpoint_api_port(self):
                 password=self.password,
             )
             _, out, _ = rgw_conn.exec_cmd(cmd)
-            port = out.split(":")[-1]
+            daemons = json.loads(out)
+            port = [
+                daemon["ports"][0]
+                for daemon in daemons
+                if daemon["service_name"].startswith("rgw")
+            ][0]
+            # if port doesn't have value, need to check followup way
+            if not port:
+                raise AttributeError(
+                    "Command `cephadm ls` output doesn't have information about rgw port."
+                )
+        except Exception as ex:
```

Comment:
Apart from AttributeError, which other exceptions could be caught here?
If it's only AttributeError that we can expect, wouldn't it be simpler to replace the try-except with conditions, such as port doesn't have value and ceph version?

Reply:
It was mostly copy&paste of the existing code extended to Ceph 6 specifics, but you are right, that it looks quite complicated and that there might be some better way how to structure the code. I'll review it and send update.

Reply:
I had to get back to the code to remind myself the functionality.
There might be various exceptions raised - basically any step in the `try .. except` block could theoretically fail (`ExternalClusterRGWEndPointMissing` - from `get_rgw_endpoint()`, connection related exceptions from `Connection` object, `KeyError` and `IndexError` and finally the `AttributeError`). In reality, there will be raised only some of those exceptions, which depends on the specifics for each Ceph cluster version (for example in older version the `cephadm` command is not available or the returned structure is different).